src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/RetryExecutor.kt

```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.hmpps.personrecord.service
+
+import kotlinx.coroutines.delay
+import kotlin.reflect.KClass
+
+private const val DELAY: Long = 2000
+
```

Comment:
```  Test should retry three times and throw the exception and send correct telemetry event() PASSED (24.1s)
  Test should track partial match event when partial match found() PASSED
  Test should retry three times before returning prisoner matcher() PASSED (4s)
  Test should not retry when exception thrown is not in the list of exceptions for retry() PASSED
  Test should add prisoner to person record when exact match found associated prisoners correctly() PASSED
  Test should retry before returning prisoner details() PASSED (6s)```
  

The tests are quite slow now, most likely  as a result of this 2s delay. 
It's not configurable at the moment, but it might be an idea to make it configurable and pass in a much smaller value for testing.
Also, the maxAttempts and exceptions are configurable but the same values are always passed in - maybe we could keep them configurable but use defaults and named arguments?

Reply:
Thought so but it is not part of the story (even the exceptions list is not part of the story).  I am ok to make the changes but I will be on holiday from tomorrow so not sure I can finish that.